docs/develop/typescript/cancellation.mdx

```diff
@@ -383,3 +383,20 @@ export async function shieldAwaitedInRootScope(): Promise<any> {
 ```
 
 <!--SNIPEND-->
+
+## Cancel an Activity from a Workflow {#cancel-an-activity}
+
+Canceling an Activity from within a Workflow requires that the Activity Execution sends Heartbeats and sets a Heartbeat Timeout.
+If the Heartbeat is not invoked, the Activity cannot receive a cancellation request.
```

Comment:
Simpler: Activities without Heartbeats can't be canceled.
Good opportunity to explain why this is necessary. I'm asking in tech-qs to see if I understand it correctly.

Reply:
> Activity cancellations are sent by the server to the worker through the RecordActivityTaskHeartbeatResponse. So if there is no heartbeat request, then there is no heartbeat response, and consequently, the worker never learns about the activity cancellation request. (edited)